spacy/pipeline/entity_linker.py

```diff
@@ -474,18 +474,29 @@ def predict(self, docs: Iterable[Doc]) -> List[str]:
 
                 # Looping through each entity in batch (TODO: rewrite)
                 for j, ent in enumerate(ent_batch):
-                    sent_index = sentences.index(ent.sent)
-                    assert sent_index >= 0
+                    sents = list(ent.sents)
+                    # Note: the last sentence associated with an sentence-crossing entity isn't complete. E. g. if you
+                    # have "Mahler's Symphony No. 8 was beautiful", the entity being "No. 8", ent.sents would be:
+                    #   1. "Mahler's Symphony No."
+                    #   2. "8"
+                    # whereas doc.sents would be:
+                    #   1. "Mahler's Symphony No."
+                    #   2. "8 was beautiful"
+                    # This makes it tricky to receive the last sentence by indexing doc.sents - hence we use an offset
+                    # to determine sent_indices[1].
+                    sent_indices = (sentences.index(sents[0]), sentences.index(sents[0]) + len(sents) - 1)
+                    assert all([si >= 0 for si in sent_indices])
```

Comment:
We could add an assertion checking that `sent_indices` is a strictly increasing sequence.